lib/rucio/db/sqla/session.py

```diff
@@ -95,26 +95,33 @@ def mysql_ping_listener(dbapi_conn, connection_rec, connection_proxy):
             raise
 
 
-def mysql_convert_decimal_to_float(dbapi_conn, connection_rec):
+def mysql_convert_decimal_to_float():
     """
     The default datatype returned by mysql-python for numerics is decimal.Decimal.
     This type cannot be serialised to JSON, therefore we need to autoconvert to floats.
     Even worse, there's two types of decimals created by the MySQLdb driver, so we must
     override both.
 
-    :param dbapi_conn: DBAPI connection
-    :param connection_rec: connection record
+    :return converter: Converter object
     """
 
+    converter = None
     try:
         import MySQLdb.converters  # pylint: disable=import-error
         from MySQLdb.constants import FIELD_TYPE  # pylint: disable=import-error
-    except:
-        raise RucioException('Trying to use MySQL without mysql-python installed!')
-    conv = MySQLdb.converters.conversions.copy()
-    conv[FIELD_TYPE.DECIMAL] = float
-    conv[FIELD_TYPE.NEWDECIMAL] = float
-    dbapi_conn.converter = conv
+        converter = MySQLdb.converters.conversions.copy()
+        converter[FIELD_TYPE.DECIMAL] = float
+        converter[FIELD_TYPE.NEWDECIMAL] = float
+    except ImportError:
+        try:
+            from pymysql.constants import FIELD_TYPE
+            from pymysql.converters import conversions as conv
+            converter = conv.copy()
+            converter[FIELD_TYPE.DECIMAL] = float
+            converter[FIELD_TYPE.NEWDECIMAL] = float
+        except ImportError:
+            raise RucioException('Trying to use MySQL without mysql-python or pymysql installed!')
```

Comment:
should we foresee dropping mysql-python at all? does pymysql work in py2? otherwise maybe adding something in parentheses would help

e.g.,  ...  without mysql-python (for Python 2) or pymysql (for Python3) ...

Reply:
pymysql should be py2 compatible, too. So we could drop mysql-python. I have read somewhere that it is faster than pymysql because pymysql is only a pure python binding, but it is easier to install via pip.